src/components/app/details/metrics/DeploymentMetrics.tsx

```diff
@@ -244,7 +251,8 @@ export default class DeploymentMetrics extends Component<DeploymentMetricsProps,
     }
 
     renderInputs() {
-        return <div className="deployment-metrics__inputs">
+      const _filteredEnvironment = this.state.environments.filter((_env) => !_env.deploymentAppDeleteRequest)
```

Comment:
Please keep the filter state along with environments as it'll be re-computed again when renderInputs renders on state changes.

Reply:
Done